### DISCUSSION TITLE
a simple request from a developer

### DISCUSSION THREAD

**Discussion by BlackImp** [EdgeTX/edgetx#4889] [Ideas]
As professional developer on different platforms and OS's with different languages, I suggest you to use less packages and languages for a simple firmware for being able to EDIT code and to compile it. I've read the deprecated Windows 10 - visual studio instructions. On one side you provide instructions for just compiling the firmware from a shell without support on how to configure any C++ IDE for using all the programs required, on the other side you provide support for windows 10 which is the environment most of the people use - and developers too - BUT it's required to installa lot of packages: sometimes I have the feelings my "collegues" have some onanistic tendency to adopt tons of different packages and languages for doing a very simple thing. For a firmware I would expect just a crosscompiler, some C/C++ libraries and LUA as you like to think users spend time learning LUA for programming their radio. what else? CMAKE? ok.
why Python 2, Python 3. It would be useful to separate packages required just for firmware than those for companion. 
By the way I don't understand why supporting now just the compilation of sources and not the editing. 
As developer I always try to semplify things for other developers and users. It seems these kind of open source projects are made with the intent of playing with different tools and packages rather than offering a code easily editable and compilable with simple instruments. Maybe it's just me and I didn't pay enough time to read build instrictions but it's a pity as having more free time at the moment I would dedicate some to a few customizations of edgetx. I prefer spending time for coding and projecting than putting together languages, packages, utilities and configuration files.

**Comment by pfeerick** [comment]
I'm not saying there isn't merit to what you are saying (as there is), but if you just want to get into the source code and work on stuff, why not just use VSCode + devcontainer (and *you* don't need to install anything then - as it it is all managed via the docker build container) or even GitPod and you don't even need to install an IDE locally? 

Yes, a lot of open source projects are simply spare time projects, and ones like this have evolved over the last 10-15 years to be what they are, which many different developers and ideas on ways things should be done. So obviously documentation is something no-one wants to work on since they're not being forced to do so as part of their job. But, at the same time, anyone can contribute, and clean the build system and documentation up ;)

**Reply by BlackImp** [reply]
thank you very much for replying me. I certainly will come back in the next days reading better the wiki and the solutions you suggested. It's understood that I won't stop here :) If I can contribute somehow on documentation I'll do it. I'll let you know.

**Reply by pfeerick** [reply]
It will be much appreciated... the documentation does need a "fresh to the project" set of eyes to try and make it as easy as possible for new folks to chip in... more the merrier! :)
